kolibri/core/assets/src/kolibri_app.js

```diff
@@ -0,0 +1,51 @@
+import KolibriModule from 'kolibri_module';
+import { getCurrentSession } from 'kolibri.coreVue.vuex.actions';
+import router from 'kolibri.coreVue.router';
+import Vue from 'kolibri.lib.vue';
+import store from 'kolibri.coreVue.vuex.store';
+
+/*
+ * A class for single page apps that control routing and vuex state.
+ * Override the routes, mutations, initialState, and RootVue getters.
+ */
+export default class KolibriApp extends KolibriModule {
+  get routes() {
+    return [];
+  }
+  get mutations() {
+    return {};
+  }
+  get initialState() {
+    return {};
+  }
+  get RootVue() {
+    return {};
+  }
+  get store() {
+    return store;
+  }
+  get stateSetters() {
+    return [];
+  }
+  ready() {
+    this.store.registerModule({
+      state: this.initialState,
+      mutations: this.mutations,
+    });
+    Promise.all([
+      getCurrentSession(store),
+      ...this.stateSetters.map(setter => setter(this.store)),
```

Comment:
I'm a bit confused about what this line does. Mind adding a comment?

Reply:
Will do.